### ISSUE TITLE
Steal Uber Movement's UI for date/time picking

### ISSUE THREAD

**Issue by hathix** [trynmaps/metrics-mvp#78]
![Screenshot 2019-04-24 at 8 04 34 PM](https://user-images.githubusercontent.com/728084/56707128-84ddca00-66cc-11e9-8549-c5577bf16757.png)

**Comment by hathix** [comment]
They do a great job adding power and flexibility to this. This way, you can ask questions like "what's the average travel time on weekdays last march?"

**Comment by exxonvaldez** [comment]
There's a bit of conumdrum when it comes to selecting a date picker UI.  You would think Material UI core would have at least a basic one (and it used to, apparently), but it doesn't.  It relies on the browser's native date picker, if any.  This isn't great -- it's inconsistent among browsers (some implement a picker, some don't), and doesn't look like the rest of the UI.

There is an active, non-core Material UI date picker called "Material UI Pickers" at https://material-ui-pickers.dev/demo/datepicker .  It doesn't support range selection, but is built to the Material UI design spec and looks good.   Range seleciton has been a frequently requested feature since 2018, but there's still no sign of that particular feature.

Uber uses its own React component library (a competitor to Material UI) called Base Web, and they use it for Uber Movement.  Their picker supports date ranges.

There's a React picker with range selection that hasn't been updated since June 2018: https://github.com/onefinestay/react-daterange-picker .  Stylistically it doesn't match Material UI.

I'm thinking that most of the time, predefined ranges should meet people's needs (last 7, 30, 90 days relative to the main date picker's date).  Those choices could be made via buttons or radio button group.  Then we can also offer a custom range UI where they use a secondary date picker to pick the starting date.

So, basically, whether we continue to use the native picker or upgrade to the Material UI Pickers one, we are forced to do a date picker UI that is not as optimal as the Uber Movement one.

**Comment by hathix** [comment]
We've stolen A LOT of it, and culturally we understand to look to Uber movement for guidance
